### PR TITLE
Upload pdf to s3

### PR DESCRIPTION
**Pull request by Cannon07**
### Changes Made:
- Updated build_pdf_book.py
- Added Unit Test script: test_build_pdf_book.py for testing the build_pdf_book.py script.
- Added Mock files in the mocks directory for running the test_build_pdf_book.py script.
- Updated the test.yml workflow to run the test_build_pdf_book.py script.
- Added build_pdf_book.yml workflow to fetch the latest version of tutorials, rebuild the book and upload to S3 bucket

### CODE REVIEW

File: new-website/utils/tutorials/build_pdf_book.py [hunk 1]
```diff
@@ -85,9 +176,11 @@ def compile_information_pages():
...
 
+
 if __name__ == "__main__":
     os.system("mkdir " + PDF_PATH)
     html_to_pdf()
     merge_pdf()
     compile_information_pages()
-    merge_pdf_pages(['storage/title.pdf', 'storage/acknowledgement.pdf', 'storage/contents.pdf', 'storage/full_pdf.pdf'])
+    merge_pdf_pages(['storage/title.pdf', 'storage/acknowledgement.pdf', 'storage/contents.pdf', 'storage/merged.pdf'])
+    upload_file('storage/full_pdf.pdf', 'deepchemtutorials', 'TutorialsBook.pdf')
```
Comment (sudo-rsingh) [inline new-website/utils/tutorials/build_pdf_book.py]: pls, keep an empy line at the end

File: new-website/utils/tutorials/test_build_pdf_book.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        build_pdf_book.merge_pdf(INFO_PATH, PDF_PATH)
+
+        self.assertTrue(os.path.isfile(
+            f"{PDF_PATH}merged.pdf"))
+        os.remove(f"{PDF_PATH}merged.pdf")
+
+
+if __name__ == "__main__":
+    unittest.main()
```
Comment (sudo-rsingh) [inline new-website/utils/tutorials/test_build_pdf_book.py]: same here, pls add an empty last line.

File: new-website/utils/tutorials/build_pdf_book.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
     - pdfunite
     - pdfkit
     - mdpdf
+    - boto3
```
Comment (sudo-rsingh) [inline new-website/utils/tutorials/build_pdf_book.py:8]: please add boto3 to requirements file

**Review by sudo-rsingh — APPROVED** [review]
LGTM, few minor fixes and good to merge.

File: .github/workflows/build_pdf_book.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: Build latest version of PDF Book
+
+on:
```
Comment (sudo-rsingh) [inline .github/workflows/build_pdf_book.yml:3]: Looks good, can you please explain when exactly this workflow will get triggered.

Comment (Cannon07) [reply .github/workflows/build_pdf_book.yml:3]: Yes, the workflow will be triggered whenever new commits are performed in examples/tutorials directory in deepchem repository.

File: new-website/utils/tutorials/test_build_pdf_book.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+"""
+
+import unittest
+from unittest.mock import patch
+import build_pdf_book
+import os
+import json
+
+
+class TestConvertHTMLToPDF(unittest.TestCase):
```
Comment (rbharath) [inline new-website/utils/tutorials/test_build_pdf_book.py:20]: Add a docstring to this class (use numpydoc style)

File: new-website/utils/tutorials/test_build_pdf_book.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    def test_convert_html_to_pdf(self):
```
Comment (rbharath) [inline new-website/utils/tutorials/test_build_pdf_book.py:25]: Add docstrings to methods

File: new-website/utils/tutorials/test_build_pdf_book.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            tutorial_file_name = tutorial["name"]
+            file_name_pdf = f'{tutorial_file_name.rsplit(".")[0]}.pdf'
+            self.assertTrue(os.path.isfile(
+                f"mocks/mock_temp_storage/{file_name_pdf}"))
+            os.remove(f"mocks/mock_temp_storage/{file_name_pdf}")
+        
+        os.rmdir("mocks/mock_temp_storage")
+            
+
+class TestMergePDF(unittest.TestCase):
```
Comment (rbharath) [inline new-website/utils/tutorials/test_build_pdf_book.py:57]: Docs for this class

**Review by rbharath — COMMENTED** [review]
Can you please update the README? Make suer to explain how these new capabilities work. Also please add docstrings in numpydocs style .

Why do we need to add mock-html notebooks? We didn't need these previously

**Comment by Cannon07** [comment]
> Can you please update the README? Make suer to explain how these new capabilities work. Also please add docstrings in numpydocs style .
> 
> Why do we need to add mock-html notebooks? We didn't need these previously

**Comment by Cannon07** [comment]
> Why do we need to add mock-html notebooks? We didn't need these previously

Since the function html_to_pdf converts the html files to pdf, we need some mock html files to unit test this functionality. So for testing that I have added mock-html notebooks folder which contains sample html-files.
We didn't need it previously because there were no tests to evaluate this functionality.

**Pull request closed without merging**